### PR TITLE
[dhcp_relay] Enhance DHCP monitor application to support DHCPv6

### PR DESCRIPTION
**Pull request by shlomibitton**
Signed-off-by: Shlomi Bitton <shlomibi@nvidia.com>

#### Why I did it

Enhance DHCP monitor application following the implementation PR: https://github.com/Azure/sonic-buildimage/pull/7772

#### How I did it

Add the support for monitoring DHCPv6 packets.

#### How to verify it

Install an image with this PR and the implementation PR.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [x] 202106

### CODE REVIEW

File: src/dhcpmon/src/dhcp_device.c [hunk 1]
```diff
@@ -152,6 +196,45 @@ static void handle_dhcp_option_53(dhcp_device_context_t *context,
...
+    case DHCPv6_MESSAGE_TYPE_CONFIRM:
+    case DHCPv6_MESSAGE_TYPE_RENEW:
+    case DHCPv6_MESSAGE_TYPE_REBIND:
+    case DHCPv6_MESSAGE_TYPE_RELEASE:
+    case DHCPv6_MESSAGE_TYPE_DECLINE:
+    case DHCPv6_MESSAGE_TYPE_ADVERTISE:
+    case DHCPv6_MESSAGE_TYPE_REPLY:
+    case DHCPv6_MESSAGE_TYPE_RECONFIGURE:
+    case DHCPv6_MESSAGE_TYPE_INFORMATION_REQUEST:
+    case DHCPv6_MESSAGE_TYPE_RELAY_FORWARD:
```
Comment (tahmed-dev) [inline src/dhcpmon/src/dhcp_device.c:227]: can we get rid of relay-forward and relay-reply message types and count inner message against DHCPv6 message type?

Comment (shlomibitton) [reply src/dhcpmon/src/dhcp_device.c:226]: @tahmed-dev I changed the code to fit your suggestion, please review.
Thanks

Comment (tahmed-dev) [reply src/dhcpmon/src/dhcp_device.c:226]: Thanks. I noticed now that the parser will point to inner type of encapped packet. Can we remove the Relay Forward and Relay Reply from this case and let them fall through to the default section as this should not be expected.

**Review by tahmed-dev — APPROVED** [review]
LGTM. just a minor comment

File: src/dhcpmon/src/dhcp_device.c (hunk 1 above)
Comment (shlomibitton) [reply src/dhcpmon/src/dhcp_device.c]: Done.

**Review by tahmed-dev — APPROVED** [review]

**Comment by lguohan** [comment]
i saw tamer approved, but now it is not in the status. @tahmed-dev, can you approve?

**Pull request merged**